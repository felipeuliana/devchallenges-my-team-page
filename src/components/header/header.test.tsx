/* eslint-disable no-useless-escape */

import React from 'react'
import { render, screen } from '@testing-library/react'
import Header from './header'

const DESCRIPTION = 'We are team of creatively diverse.\s\sdriven.\s\sinnovative individuals working in various loctions from the world.'
const HEADING = 'The creative crew'
const SUBTITLE = 'Who we are'

describe('Header component', () => {
  let component: HTMLElement

  beforeEach(() => {
    const { container } = render(
      <Header
        description={DESCRIPTION}
        heading={HEADING}
        subtitle={SUBTITLE}
      />)

    component = container
  })

  it('should render correctly', () => {
    expect(component).toMatchSnapshot()
  })

  it('should display the heading', () => {
    const heading = screen.getByRole(
      'heading', { level: 1, name: /the creative crew/i }
    )

    expect(heading).toBeInTheDocument()
  })

  it('should display the subtitle', () => {
    const subtitle = screen.getByRole(
      'heading', { level: 2, name: /who we are/i }
    )

    expect(subtitle).toBeInTheDocument()
  })

  it('should display the description', () => {
    const description = screen.getByText(DESCRIPTION)

    expect(description).toBeInTheDocument()
  })
})
