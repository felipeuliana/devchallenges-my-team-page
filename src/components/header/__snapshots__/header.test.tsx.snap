// Vitest Snapshot v1

exports[`Header component > should render correctly 1`] = `
<div>
  <header
    class="_root_503ab9"
  >
    <div
      class="_inner_503ab9"
    >
      <h1
        class="_heading_503ab9"
      >
        The creative crew
      </h1>
    </div>
    <div
      class="_inner_503ab9"
    >
      <h2
        class="_subtitle_503ab9"
      >
        Who we are
      </h2>
      <p
        class="_description_503ab9"
      >
        We are team of creatively diverse.ssdriven.ssinnovative individuals working in various loctions from the world.
      </p>
    </div>
  </header>
</div>
`;
